lib/utils.js

```diff
@@ -7,98 +10,79 @@ var assert = require("assert");
-    }
+    return pathname ? pathname.replace('/', '') : pathname;
   }
 
-  var host_port;
-  if (hostList.length > 1) {
-    result.database.host = [];
-    result.database.port = [];
-    for (i = 0; i < hostList.length; i++) {
-      host_port = hostList[i].split(":");
-      assert.ok(host_port[0] && host_port[0].length > 0);
-      result.database.host.push(host_port[0]);
-      if (host_port.length > 1) {
-        result.database.port.push(host_port[1]);
-      } else {
-        result.database.port.push('27017');
-      }
+  // Parse the query string from the last host portion
+  function getOptionsObject () {
+    if (parts.length > 1) {
+      // Options should be appended to the last hosts querystring
+      return qs.parse(url.parse(parts[parts.length - 1]).query);
     }
+
+    return {};
   }
-  else {
-    host_port = hostList[0].split(":");
-    assert.ok(host_port[0] && host_port[0].length > 0);
-    result.database.host = host_port[0];
-    if (host_port.length > 1) {
-      result.database.port = host_port[1];
+  
+  // Authentication is defined in the primary host URL
+  function getAuthenticationObject () {
+    const authString = url.parse(parts[0]).auth;
+
+    if (authString) {
+      return {
+        source: getName(),
+        user: authString.split(':')[0],
+        pass: authString.split(':')[1]
+      };
     } else {
-      result.database.port = '27017';
+      return {};
     }
   }
 
+  // Only the primary host string contains the "mongodb://" so we use this
+  // function to prepend it to others when passing to url.parse
+  function prependProtocol (str) {
+    if (str.indexOf('mongodb://') === -1) {
+      return 'mongodb://' + str;
+    } else {
+      return str;
+    }
+  }
 
-  //Verify the config to be returned
-  assert.ok(result.database.name.length > 0);
-  assert.ok(result.database.auth.user.length > 0);
-  assert.ok(result.database.auth.pass.length > 0);
-
-
-  return result;
+  var result = {
+    database: {
+      driver_options: getOptionsObject(),
+      auth: getAuthenticationObject(),
+      name: getName(),
+      
+      // Use map since we can have multiple hosts and ports
+      host: parts.map((p) => url.parse(prependProtocol(p)).hostname),
+      port: parts.map((p) => url.parse(p).port || 27017)
+    }
+  };
+
+  // Verify a database name is given in the connection string
+  assert.ok(result.database.name, 'database name must be defined, e.g mongodb://localhost:27017/DATABASE_NAME');
+
+  // At least one host and port must exist
+  assert.notEqual(result.database.host, 0, 'at least one host must be specified in the mongodb connection string');
+  assert.notEqual(result.database.port, 0, 'at least one host must be specified in the mongodb connection string');
+  
+  if (!env('FH_USE_LOCAL_DB').asBool()) {
+    // We can't assume authentication is being used if run locally
+    assert.ok(result.database.auth.user, 'mongodb username was missing in connection string');
```

Comment:
@camilamacedo86 yep, `assert` is usually used in test cases, but is also valid to use as a "fail fast" mechanism in scenarios like this. Since this is the DB logic we want our app startup to fail of we have a bad connection string so we can fix it, otherwise the app will deploy and break later - not good!